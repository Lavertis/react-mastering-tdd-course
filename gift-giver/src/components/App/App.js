import {Component} from "react";
import {Button, Col} from "react-bootstrap";
import Gift from "../Gift/Gift";
import {maxNumber} from "../../helpers";

class App extends Component {
    constructor() {
        super();

        this.state = {gifts: []};
    }

    addGift = () => {
        const {gifts} = this.state;
        const ids = gifts.map(gift => gift.id);
        const maxId = maxNumber(ids);
        gifts.push({id: maxId + 1});
        this.setState({gifts});
    };

    removeGift = (id) => {
        const gifts = this.state.gifts.filter(gift => gift.id !== id);
        this.setState({gifts});
    };

    render() {
        return (
            <Col xs={4} className="App text-center mx-auto my-5">
                <h2>Gift Giver</h2>
                <div className="gift-list">
                    {this.state.gifts.map(gift => {
                        return (
                            <Gift
                                key={gift.id}
                                gift={gift}
                                removeGift={this.removeGift}
                            />
                        );
                    })}
                </div>
                <Button className="btn-add m-2" onClick={this.addGift}>Add Gift</Button>
            </Col>
        );
    }
}

export default App;
