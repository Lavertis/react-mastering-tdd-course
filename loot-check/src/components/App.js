function App() {
    return (
        <div className="App">
            <h2>Loot Check</h2>
        </div>
    );
}

export default App;
