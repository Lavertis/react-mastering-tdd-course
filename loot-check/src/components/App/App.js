import Wallet from "../Wallet/Wallet";
import Loot from "../Loot/Loot";

function App() {
    return (
        <div className="App">
            <h2>Loot Check</h2>
            <hr/>
            <Wallet/>
            <hr/>
            <Loot/>
            <div>
                Powered by&nbsp;
                <a target="_blank" rel="noreferrer" href="https://www.coindesk.com/price">Coindesk</a>
            </div>
        </div>
    );
}

export default App;
