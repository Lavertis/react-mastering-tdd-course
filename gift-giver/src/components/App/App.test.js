import React from "react";
import App from './App';
import {shallow} from "enzyme";

const app = shallow(<App/>)

it('renders correctly', () => {
    expect(app).toMatchSnapshot();
});

it('initializes the `state` with a empty list of gifts', () => {
    jest.spyOn(React, 'useState')
    expect(app.state().gifts).toEqual([])
});

it('adds a new gift to `state` when clicking the `add gift` button', () => {
    app.find('.btn-add').simulate('click');
    expect(app.state().gifts).toEqual([{id: 1}]);
});

it('adds a new gift to the rendered list when clicking the `add gift` button', () => {
    app.find('.btn-add').simulate('click');
    expect(app.find('.gift-list').children()).toHaveLength(2);
})