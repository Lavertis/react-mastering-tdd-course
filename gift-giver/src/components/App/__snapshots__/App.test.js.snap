// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <h2>
    Gift Giver
  </h2>
  <Button
    active={false}
    className="btn-add"
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    Add Gift
  </Button>
</div>
`;
